agent/consul/leader_routine_manager_test.go

```diff
@@ -16,7 +16,7 @@ func TestLeaderRoutineManager(t *testing.T) {
 	var running uint32
 	// tlog := testutil.NewCancellableTestLogger(t)
 	// defer tlog.Cancel()
-	mgr := NewLeaderRoutineManager(testutil.TestLogger(t))
+	mgr := NewLeaderRoutineManager(testutil.TestHcLog(t.Name()))
```

Comment:
The important part of `testutil.TestLogger` isn't the extraction of the test name, but rather the fact that under the covers the `testing.TB.Logf` function is massaged into a `io.Writer` and forms the logsink. This ensures that all of the log messages emitted by a test case are funneled through the `Logf` function that stops working (by design) when the test concludes. This is super helpful for preventing log message bleedover between unrelated test cases.